sysbuild/Kconfig.suit

```diff
@@ -31,14 +31,75 @@ config SUIT_ENVELOPE_ROOT_TEMPLATE_FILENAME
 config SUIT_ENVELOPE_SIGN_SCRIPT
 	string "Location of SUIT sign script"
 	depends on SUIT_ENVELOPE_SIGN
+	default "modules/lib/suit-generator/ncs/sign_script.py"
 	help
 	  Python script called to sign SUIT envelope.
 	  You can use either absolute or relative path.
 	  In case relative path is used, the build system uses NRF parent directory.
 	  Script need to accept two arguments:
 	  - --input-file <STRING> - location of unsigned envelope in the build system
 	  - --output-file <STRING> - location of signed envelope to create by script
-	default "modules/lib/suit-generator/ncs/sign_script.py"
+
+choice SUIT_ENVELOPE_ENCRYPT_SCRIPT
+	prompt "Select the used SUIT encryption script"
+	default SUIT_ENVELOPE_ENCRYPT_SCRIPT_DEFAULT
+	help
+	   The SUIT encryption script is used to generate encryption artifacts for images inside a SUIT envelope.
+	   It is the "external" script that is called by the build system - it is responsible for creating
+	   various needed SUIT structures, but passes the actual encryption of the payload to a separate KMS script.
+
+config SUIT_ENVELOPE_ENCRYPT_SCRIPT_DEFAULT
+	bool "Use the default SUIT encryption script"
+	help
+	  Use the default encryption script to generate encryption artifacts for images inside a SUIT envelope.
+
+endchoice
+
+config SUIT_ENVELOPE_ENCRYPT_SCRIPT_PATH
+	string "Location of SUIT encrypt script"
+	default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/encrypt_script.py" if SUIT_ENVELOPE_ENCRYPT_SCRIPT_DEFAULT
+	help
+	  Python script called to generate encryption artifacts for images inside a SUIT envelope.
+	  See the help message for the default encryption script to see what arguments the script
+	  must accept.
+
+choice SUIT_ENVELOPE_KMS_SCRIPT
+	prompt "Select the used SUIT KMS script to be used by the encryption script"
+	help
+	   This is the "internal" python script that is called by the encryption script to perform the cryptographic operations,
+	   such as encryption using an AES key. The script must contain a class implementing the interface defined by
+	   the abstract class SuitKMSBase found in the suit_generator.suit_kms_base and a suit_kms_factory function
+	   needed to get an object of the class.
+
+config SUIT_ENVELOPE_KMS_SCRIPT_BASIC
+	bool "Use the basic SUIT KMS script"
+	help
+	  Use the basic KMS script to encrypt firmware.
+	  This script provides a basic implementation of the SuitKMS interface, in which
+	  the private/encryption keys are stored in the filesystem.
+
+endchoice
+
+config SUIT_ENVELOPE_KMS_SCRIPT_PATH
+	string "Location of SUIT KMS script"
+	default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/basic_kms.py" if SUIT_ENVELOPE_KMS_SCRIPT_BASIC
```

Comment:
why is this a kconfig?

Reply:
The `SUIT_ENVELOPE_KMS_SCRIPT` choice is supposed to be extended, adding more possible values to `SUIT_ENVELOPE_KMS_SCRIPT_PATH` - in a similar way like https://github.com/nrfconnect/sdk-nrf/blob/cac49ac985664e9357e1c30691ce1091470c528b/sysbuild/Kconfig.netcore#L142

Reply:
The netcore part is used to specify an image, not a file, the file part of signing is done like so: https://github.com/zephyrproject-rtos/zephyr/blob/main/CMakeLists.txt#L2027 which then uses this file by default: https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/mcuboot.cmake but in ncs is overridden here: https://github.com/nrfconnect/sdk-nrf/blob/main/sysbuild/CMakeLists.txt#L293 or https://github.com/nrfconnect/sdk-nrf/blob/main/sysbuild/CMakeLists.txt#L336

Reply:
These CMake parts are completely unrelated to SUIT. We cannot use any of them, as they are strongly coupled in the way MCUBoot works - it has a completely different structure then the whole SUIT Cmake build system - the "signing scripts" in mcuboot are actually doing a lot of things  which is done by the https://github.com/nrfconnect/sdk-nrf/blob/main/cmake/sysbuild/suit.cmake - in SUIT, all the scripts related to signing operate at a much lower level.

Reply:
Also note - this is not a CMake script - this is a python script. It operates in level closer to something like imgtool in mcuboot, even lower.

Reply:
Ok, the misunderstanding was cleared offline, I will fix this.